-- Warning: tests/neg-custom-args/captures/byname.scala:5:21 -----------------------------------------------------------
5 |  def g(x: Int) = if cap2 == cap2 then 1 else x
  |                     ^^^^
  |                     Another capture checking run needs to be scheduled because
  |                     reference (cap2 : Cap) is not included in the previously estimated
  |                     capture set {} of method f
-- [E007] Type Mismatch Error: tests/neg-custom-args/captures/byname.scala:10:6 ----------------------------------------
10 |  h(f2())  // error
   |    ^^^^
   |Found:    () ?->{cap1} Int ->{cap1} Int
   |Required: () ?=> Int ->{cap2} Int
   |
   |where:    ?=> refers to a fresh root capability created in method test when checking argument to parameter ff of method h
   |
   |Note that capability (cap1 : Cap) is not included in capture set {cap2}.
   |
   | longer explanation available when compiling with `-explain`
-- [E007] Type Mismatch Error: tests/neg-custom-args/captures/byname.scala:19:5 ----------------------------------------
19 |  h(g()) // error
   |    ^^^
   |    Found:    () ?->{cap2} I^?
   |    Required: () ?->{cap1} I
   |    Note that capability (cap2 : Cap) is not included in capture set {cap1}.
   |
   | longer explanation available when compiling with `-explain`
-- [E007] Type Mismatch Error: tests/neg-custom-args/captures/byname.scala:22:5 ----------------------------------------
22 |  h2(() => g())() // error
   |     ^^^^^^^^^
   |     Found:    () ->{cap2} I^?
   |     Required: () ->{cap1} I
   |     Note that capability (cap2 : Cap) is not included in capture set {cap1}.
   |
   | longer explanation available when compiling with `-explain`
